### PR TITLE
feat(template): support optional parameters

### PR DESCRIPTION
**Pull request by rubenfonseca**
*Issue #, if available:* #542

*Description of changes:*

Marks optional fields as pointers so we can serialize/deserialize with confidence.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by rclark** [comment]
:+1: Yep that solves my problem. Thank you!

File: cloudformation/template.go [hunk 1]
```diff
@@ -25,24 +25,24 @@ type Template struct {
...
-	MaxLength             int           `json:"MaxLength,omitempty"`
-	MinLength             int           `json:"MinLength,omitempty"`
-	MaxValue              float64       `json:"MaxValue,omitempty"`
-	MinValue              float64       `json:"MinValue,omitempty"`
-	NoEcho                bool          `json:"NoEcho,omitempty"`
+	Type                  string         `json:"Type"`
+	Description           *string        `json:"Description,omitempty"`
+	Default               *interface{}   `json:"Default,omitempty"`
+	AllowedPattern        *string        `json:"AllowedPattern,omitempty"`
+	AllowedValues         *[]interface{} `json:"AllowedValues,omitempty"`
```
Comment (xrn) [inline cloudformation/template.go:32]: There is no need to make a pointer on [] in golang as far as i know

**Comment by xrn** [comment]
There is no need to make a pointer on [] in golang as far as i know `AllowedValues         *[]interface{}`

**Comment by rubenfonseca** [comment]
@xrn you're absolutely right. Can you please check if it makes sense now?

**Comment by xrn** [comment]
Looks great

**Pull request merged**